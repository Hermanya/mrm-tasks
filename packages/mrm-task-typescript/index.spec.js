/* eslint-disable no-console */
'use strict';

jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));
jest.mock('mrm-core/src/npm', () => ({
	install: jest.fn(),
}));

const { getConfigGetter } = require('mrm');
const vol = require('memfs').vol;
const task = require('./index');

const json = o => JSON.stringify(o, null, '  ');

const packageJson = json({
	name: 'unicorn',
	scripts: {
		test: 'jest',
	},
});

afterEach(() => vol.reset());

it('should add TypeScript', () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(getConfigGetter({}));

	expect(vol.toJSON()).toMatchSnapshot();
});
