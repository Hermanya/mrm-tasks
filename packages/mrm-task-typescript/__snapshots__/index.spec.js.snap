// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add TypeScript 1`] = `
Object {
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"test\\": \\"jest\\",
    \\"pretest\\": \\"tsc --noEmit\\"
  }
}",
  "/tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"target\\": \\"es6\\",
    \\"module\\": \\"commonjs\\",
    \\"moduleResolution\\": \\"node\\",
    \\"strict\\": true,
    \\"experimentalDecorators\\": true,
    \\"emitDecoratorMetadata\\": true,
    \\"noUnusedLocals\\": true,
    \\"pretty\\": true,
    \\"lib\\": [
      \\"esnext\\"
    ]
  }
}",
}
`;
